src/main/java/com/glencoesoftware/omero/ms/core/OmeroWebJDBCSessionStore.java

```diff
@@ -0,0 +1,210 @@
+        }
+    }
+
+    private IConnector getConnectorFromSessionData(String sessionData){
+        String decodedSessionData =
+            StringUtil.fromBytes(Base64.getDecoder().decode(sessionData));
+        PyString pystring = Py.newString(decodedSessionData);
+        PyList hash_and_data = pystring.split(":", 1);
+        PyString data_str = new PyString((String) hash_and_data.get(1));
+        PyDictionary djangoSession =
+            (PyDictionary) cPickle.loads(data_str);
+        log.debug("Session: {}", djangoSession);
+        IConnector connector = (IConnector) djangoSession.get("connector");
+        return connector;
+    }
+
+    /**
+     * Retrieve the OMERO.web session's current
+     * <code>omeroweb.connector.Connector</code> synchronously.
+     * @param sessionKey Session key to retrieve a connector for.
+     * @return Connector instance or <code>null</code> if session lookup fails.
+     */
+    public IConnector getConnector(String sessionKey){
+        PreparedStatement st = null;
+        try {
+            st = sync_connection.prepareStatement(SELECT_SESSION_SQL);
+            st.setString(1, sessionKey);
+            java.sql.ResultSet rs = st.executeQuery();
+            if (!rs.next()){
+                //Nothing returned from query
+                return null;
+            }
+            else{
+                String sessionData = rs.getString(1);
+                return getConnectorFromSessionData(sessionData);
+            }
+        }
+        catch (SQLException e){
+        }
+        finally {
+            try{
+                st.close();
+            }
+            catch (SQLException e){
+                log.error(e.toString());
+            }
+        }
+
+        
+        return null;
+    }
+
+    /**
+     * Retrieve the OMERO.web session's current
+     * <code>omeroweb.connector.Connector</code>.
+     * @param sessionKey Session key to retrieve a connector for.
+     * @return A new {@link CompletionStage} that, when the {@link IConnector}
+     * retrieval is complete is executed.
+     */
+    public CompletionStage<IConnector> getConnectorAsync(String sessionKey){
+        CompletableFuture<IConnector> promise = new CompletableFuture<IConnector>();
+        client.getConnection(conn -> {
+            if (conn.failed()) {
+                log.error(conn.cause().getMessage());
+                return;
+            }
+            final SQLConnection connection = conn.result();
+
+            connection.queryWithParams(SELECT_SESSION_SQL,
+                new JsonArray().add(sessionKey), rs -> {
+                if (rs.failed()) {
+                    log.error("Cannot retrieve data from the database");
+                    rs.cause().printStackTrace();
+                    promise.complete(null);
+                    connection.close(done -> {
+                        if (done.failed()) {
+                            throw new RuntimeException(done.cause());
```

Comment:
Need to look at this strategy vs. the other implementations. Will come back to it. Pretty sure you want to fail the promise instead with the exception.